File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chat_#{params[:room]}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Message.create!(
      content: data['message'],
      sender_id: data['sender_id'],
      sender_type: data['sender_type'],
      receiver_id: data['receiver_id']
    )
    ActionCable.server.broadcast("chat_#{params[:room]}", message: data['message'])
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

consumer.subscriptions.create({ channel: "ChatChannel", room: "room_name" }, {
  connected() {
    console.log("Connected to the chat room")
  },

  disconnected() {
    console.log("Disconnected from the chat room")
  },

  received(data) {
    console.log("Received message: ", data.message)
    // Ajoute ici la logique pour afficher le message dans le chat
  },

  speak(message, senderId, senderType, receiverId) {
    this.perform('speak', { message: message, sender_id: senderId, sender_type: senderType, receiver_id: receiverId })
  }
});
